### PR TITLE
Adding a glean extra key in case of handshake errors about the Controller operation - VPN-4083

### PR DESCRIPTION
**Pull request by bakulf**
@brizental can you review the code? I still need to ask for a data-review, but first I want to know if we are OK with this approach.

### CODE REVIEW

File: glean/metrics.yaml [hunk 1]
```diff
@@ -949,6 +949,11 @@ sample:
       transport:
         description: The Host network type (i.e wifi or 4g)
         type: string
+      operation_type:
+        description: |
+          The type of operation the controller was performing. Ex. none,
+          silent-switch, switch
```
Comment (brizental) [inline glean/metrics.yaml:955]: Can't we extract this same information from: https://dictionary.telemetry.mozilla.org/apps/mozilla_vpn/metrics/sample_controller_step?

What exactly is it we want to answer with this new collection?

File: vpnglean/glean_parser_ext/templates/macros.jinja2 [hunk 2]
```diff
@@ -29,7 +29,7 @@ struct {{ obj.name|Camelize }}ExtraParser : EventMetricExtraParser {
       keepStringsAlive.append(s);
       ffiExtras.values[count] = s.constData();
       {% elif type == "boolean" %}
-      ffiExtras.values[count] = parsedExtras._{{item|camelize}} ? "true" : "false";
+      ffiExtras.values[count] = parsedExtras._{{item|camelize}}.toBool() ? "true" : "false";
```
Comment (brizental) [inline vpnglean/glean_parser_ext/templates/macros.jinja2:32]: I am not sure the new collection is necessary. However, this definitely is :D Thanks for catching it!

File: glean/metrics.yaml (hunk 1 above)
Comment (bakulf) [reply glean/metrics.yaml:955]: We want to collect errors, right? Some errors are generated when we are in StateOn, which could come from StateSwitching.

File: vpnglean/glean_parser_ext/templates/macros.jinja2 (hunk 2 above)
Comment (bakulf) [reply vpnglean/glean_parser_ext/templates/macros.jinja2:32]: Initially I used a boolean :)

File: glean/metrics.yaml (hunk 1 above)
Comment (brizental) [reply glean/metrics.yaml:955]: Sure, but we can infer the state at the time the error occured by looking at the other event I linked to there.

Comment (bakulf) [reply glean/metrics.yaml:955]: Yes, but it's not easy to compute I guess. The handshake timeout could happen in Connecting or Confirming state and we then need to go back in time to see the sequence of State changes per ping. Is it doable? Is it better to keep the current PR?

Comment (brizental) [reply glean/metrics.yaml:955]: It is doable. It's not hard, it's medium difficulty. It requires some SQL or Looker calculations, but we should be able to do that. I am not very much in favour of adding redundant collection.

**Pull request closed without merging**